File: C#/2583 - Chirrin Chirrion.cs
/// author Josélio de S. C. Júnior <joseliojrx25@gmail.com>
/// copyright Josélio de S. C. Júnior 2022

using System;
using System.Linq;
using System.Collections.Generic;

class URI {
  static void Main(string[] args) { 
      int n1 = int.Parse(Console.ReadLine() ?? "1");
  
      for (int i = 0; i < n1; i++) {
      int n2 = int.Parse(Console.ReadLine() ?? "1");
      var iv = new List<string>();
  
      for (int j = 0; j < n2; j++) {
        string[] cmd = (Console.ReadLine() ?? "").Split(' ');
  
        bool exist = iv.Exists(e => e == cmd[0]);
  
        if (cmd[1] == "chirrin" && !exist) iv.Add(cmd[0]);
        if (cmd[1] == "chirrion" && exist) iv.Remove(cmd[0]);
      }
  
      iv = iv.OrderBy(e => e).ToList();
  
      Console.WriteLine($"TOTAL\n{string.Join("\n", iv)}");
    }
  }
}

File: Python/2583 - Chirrin Chirrion.py
# author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
# copyright: Josélio Júnior (Lunatic Fox) 2022

tn = int(input())

for _ in range(tn):
  inv = []
  ut = int(input())

  for i in range(ut):
    stuff, cmd = input().split(' ')

    if (stuff not in inv) and cmd == 'chirrin':
      inv.append(stuff)
    elif (stuff in inv) and cmd == 'chirrion':
      inv.remove(stuff)

  inv.sort()
  inv = '\n'.join(inv)
  print(f'TOTAL\n{inv}')
